applications/callflow/src/module/cf_collect_dtmf.erl

```diff
@@ -78,6 +79,10 @@ collect_more_digits(Data, Call, AlreadyCollected, MaxDigits) ->
             lager:debug("failed to collect DTMF: ~p", [_E])
     end.
 
+-spec truncate_after_terminator(binary(), ne_binaries()) -> binary().
+truncate_after_terminator(AlreadyCollected, Terminators) ->
+    hd(binary:split(AlreadyCollected, Terminators, ['global'])).
```

Comment:
I don't think you need global here, as I think you just want the first sequence of DTMF that doesn't contain a terminator. For instance:

```erlang
(pdx@pdx.2600hz.com)2> binary:split(<<"1234#456#789">>, [<<"#">>, <<"*">>]).                                                                                                                                       
[<<"1234">>,<<"456#789">>]                                                                                                                                                                                         
(pdx@pdx.2600hz.com)3> binary:split(<<"1234#456#789">>, [<<"#">>, <<"*">>], [global]).                                                                                                                             
[<<"1234">>,<<"456">>,<<"789">>] 
```
`global` requires traversing the whole binary looking for terminators; omitting it ends execution once a terminator is found.